gateway-ha/src/main/java/io/trino/gateway/ha/handler/ProxyUtils.java

```diff
@@ -0,0 +1,164 @@
+    {
+        if (!isNullOrEmpty(userHeader)) {
+            log.debug("user from %s", USER_HEADER);
+            return userHeader;
+        }
+
+        log.debug("user from basic auth");
+        String user = "";
+        if (authorization == null) {
+            log.debug("didn't find any basic auth header");
+            return user;
+        }
+
+        int space = authorization.indexOf(' ');
+        if ((space < 0) || !authorization.substring(0, space).equalsIgnoreCase("basic")) {
+            log.error("basic auth format is incorrect");
+            return user;
+        }
+
+        String headerInfo = authorization.substring(space + 1).trim();
+        if (isNullOrEmpty(headerInfo)) {
+            log.error("The encoded value of basic auth doesn't exist");
+            return user;
+        }
+
+        String info = new String(Base64.getDecoder().decode(headerInfo));
+        List<String> parts = Splitter.on(':').limit(2).splitToList(info);
+        if (parts.size() < 1) {
+            log.error("no user inside the basic auth text");
+            return user;
+        }
+        return parts.get(0);
+    }
+
+    public static String extractQueryIdIfPresent(HttpServletRequest request)
+    {
+        String path = request.getRequestURI();
+        String queryParams = request.getQueryString();
+        try {
+            String queryText = CharStreams.toString(new InputStreamReader(request.getInputStream()));
+            if (!isNullOrEmpty(queryText)
+                    && queryText.toLowerCase().contains("system.runtime.kill_query")) {
+                // extract and return the queryId
+                String[] parts = queryText.split(",");
+                for (String part : parts) {
+                    if (part.contains("query_id")) {
+                        Matcher m = EXTRACT_BETWEEN_SINGLE_QUOTES.matcher(part);
+                        if (m.find()) {
+                            String queryQuoted = m.group();
+                            if (!isNullOrEmpty(queryQuoted) && queryQuoted.length() > 0) {
+                                return queryQuoted.substring(1, queryQuoted.length() - 1);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        catch (Exception e) {
+            log.error(e, "Error extracting query payload from request");
+        }
+
+        return extractQueryIdIfPresent(path, queryParams);
+    }
+
+    public static String extractQueryIdIfPresent(String path, String queryParams)
+    {
+        if (path == null) {
+            return null;
+        }
+        String queryId = null;
+
+        log.debug("trying to extract query id from  path [%s] or queryString [%s]", path, queryParams);
+        if (path.startsWith(V1_STATEMENT_PATH) || path.startsWith(V1_QUERY_PATH)) {
+            String[] tokens = path.split("/");
+            if (tokens.length >= 4) {
+                if (path.contains("queued")
```

Comment:
are these available as constants in Trino somehow?

Reply:
https://github.com/trinodb/trino/blob/6c9736e3969e774ca77e15748e7e933cc54de6a7/core/trino-main/src/main/java/io/trino/server/protocol/Query.java#L634-L655
I don't think so.

Reply:
Sad .. but I guess we will leave it at that for now.